api/controllers/game/resolve.js

```diff
@@ -135,17 +135,15 @@ module.exports = function (req, res) {
 
                 //Replace top card, if there's a card in deck
                 if (game.deck.length >= 1) {
-                  const newTopCard = _.sample(game.deck).id;
+                  const newTopCardIndex = _.random(game.deck.length - 1);
+                  const newTopCard = game.deck.splice(newTopCardIndex, 1)[0].id;
```

Comment:
Picking the card via a random index and then removing the selected card from `game.deck` in memory like this lets us select the second card with confidence it won't redundantly collide with the new topCard. This prevents the need for the `while` loop resampling